src/app/reporting/Engine.cpp

```diff
@@ -275,14 +294,15 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
         reportDataBuilder.SubscriptionId(subscriptionId);
     }
 
+    SuccessOrExit(err = reportDataWriter.ReserveBuffer(Engine::kReservedSizeForMoreChunksFlag));
```

Comment:
What about other reservations needed to close out the array of attribute report IBs?

Reply:
IIRC, They are done by StartContainer and FinishContainer